scipy/stats/_continuous_distns.py

```diff
@@ -678,33 +678,62 @@ def _pdf(self, x, a, b):
         # beta.pdf(x, a, b) = ------------------------------------
         #                              gamma(a)*gamma(b)
         with np.errstate(over='ignore'):
-            return _boost._beta_pdf(x, a, b)
+            return scu._beta_pdf(x, a, b)
 
     def _logpdf(self, x, a, b):
         lPx = sc.xlog1py(b - 1.0, -x) + sc.xlogy(a - 1.0, x)
         lPx -= sc.betaln(a, b)
         return lPx
 
     def _cdf(self, x, a, b):
-        return _boost._beta_cdf(x, a, b)
+        return sc.betainc(a, b, x)
 
     def _sf(self, x, a, b):
-        return _boost._beta_sf(x, a, b)
+        return scu._beta_sf(x, a, b)
 
     def _isf(self, x, a, b):
-        with np.errstate(over='ignore'):  # see gh-17432
-            return _boost._beta_isf(x, a, b)
+        return sc.betaincinv(a, b, 1 - x)
 
     def _ppf(self, q, a, b):
-        with np.errstate(over='ignore'):  # see gh-17432
-            return _boost._beta_ppf(q, a, b)
+        return sc.betaincinv(a, b, q)
```

Comment:
So something related to this change. I think PPF of beta distribution is inverse regularised beta incomplete function. However, `_boost._beta_ppf` gives different answers from `sc.betaincinv` for one particular case. Shouldn't they both be giving same behaviour and same answers? I know `_boost._beta_ppf` uses a more general algorithm that is I think it actually takes the inverse of the CDF instead of using the direct result. However, mathematically AFAIK, `sc.betaincinv` should be same as `_boost._beta_ppf`. Am I missing something here?

**`_boost._beta_ppf`**

```zsh
Python 3.11.8 | packaged by conda-forge | (main, Feb 16 2024, 20:49:36) [Clang 16.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from scipy import stats
>>> q, a, b = 0.995, 1.0e11, 1.0e13
>>> stats.beta.ppf(q, a, b)
/Users/czgdp1807/Quansight/scipy/build-install/lib/python3.11/site-packages/scipy/stats/_continuous_distns.py:699: RuntimeWarning: Error in function boost::math::tgamma<d>(%1%,%1%): Series evaluation exceeded %1% iterations, giving up now.
  return _boost._beta_ppf(q, a, b)
/Users/czgdp1807/Quansight/scipy/build-install/lib/python3.11/site-packages/scipy/stats/_continuous_distns.py:699: RuntimeWarning: overflow encountered in _beta_ppf
  return _boost._beta_ppf(q, a, b)
/Users/czgdp1807/Quansight/scipy/build-install/lib/python3.11/site-packages/scipy/stats/_continuous_distns.py:699: RuntimeWarning: invalid value encountered in _beta_ppf
  return _boost._beta_ppf(q, a, b)
0.009901070347291008
>>> exit()
```

``**sc.betaincinv**``

```zsh
Python 3.11.8 | packaged by conda-forge | (main, Feb 16 2024, 20:49:36) [Clang 16.0.6 ] on darwin
Type "help", "copyright", "credits" or "license" for more information.
(InteractiveConsole)
>>> from scipy import stats
>>> q, a, b = 0.995, 1.0e11, 1.0e13
>>> stats.beta.ppf(q, a, b)
nan
```

Reply:
CC @WarrenWeckesser : do you maybe have an idea about this difference?